From: Tommy Webb <tommy@calyxinstitute.org>
Date: Thu, 9 May 2024 21:17:30 +0000
Subject: Disable WebView field trials

Commit 184cb4d33787fbd7069c4ed195ba4a89df9a3207 removed the flag we
used to use to turn these off (kWebViewOriginTrials) in the
"Disable fetching of all field trials" change. We now do this in
a separate change since upstream Cromite does not currently
build WebView.

Change-Id: I92a6d6cb7f3e306a11730e4b734d764debfe57d5
---
 .../browser/component_updater/origin_trials_component_loader.cc | 2 +-
 android_webview/common/aw_content_client.cc                     | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/android_webview/browser/component_updater/origin_trials_component_loader.cc b/android_webview/browser/component_updater/origin_trials_component_loader.cc
--- a/android_webview/browser/component_updater/origin_trials_component_loader.cc
+++ b/android_webview/browser/component_updater/origin_trials_component_loader.cc
@@ -14,7 +14,7 @@ namespace android_webview {
 
 void LoadOriginTrialsComponent(
     component_updater::ComponentLoaderPolicyVector& policies) {
-  policies.push_back(std::make_unique<OriginTrialsComponentLoaderPolicy>());
+  // do nothing in Cromite
 }
 
 }  // namespace android_webview
diff --git a/android_webview/common/aw_content_client.cc b/android_webview/common/aw_content_client.cc
--- a/android_webview/common/aw_content_client.cc
+++ b/android_webview/common/aw_content_client.cc
@@ -115,6 +115,7 @@ void AwContentClient::ExposeInterfacesToBrowser(
 }
 
 blink::OriginTrialPolicy* AwContentClient::GetOriginTrialPolicy() {
+  if ((true)) return nullptr;  // do nothing in Cromite
   // Prevent initialization race (see crbug.com/721144). There may be a
   // race when the policy is needed for worker startup (which happens on a
   // separate worker thread).
-- 

